src/coprocessor/endpoint.rs

```diff
@@ -932,12 +940,12 @@ mod tests {
 
         let mut req = Request::new();
         req.set_tp(REQ_TYPE_DAG);
-        let (tx, rx) = mpsc::channel();
-        let on_resp = OnResponse::Unary(box move |msg| tx.send(msg).unwrap());
+        let (tx, rx) = oneshot::channel();
+        let on_resp = OnResponse::Unary(tx);
         let task = RequestTask::new(req, on_resp, 1000).unwrap();
 
         worker.schedule(Task::Request(task)).unwrap();
-        let resp = rx.recv_timeout(Duration::from_secs(3)).unwrap();
+        let resp = rx.wait().unwrap();
```

Comment:
It's bad for a test that doesn't have a timeout, which can make the case run forever and probably be killed by CI. If the cases are run concurrently then you can hardly tell which case is blocking the tests.

Reply:
It seems that future doesn't provide something like `wait_timeout`. Suggestions?

Reply:
You can use `select(tokio_timer.timeout)`

Reply:
I see. However I found out that there are already buckets of `.wait()` without a timeout in current repo, mainly in futurespool and storage. I think it would be better to open another PR to deal with them together, since we may need a helper to reduce code duplication in implementing this kind of `wait_timeout`.